src/main/java/org/owasp/esapi/logging/log4j/Log4JLoggerFactory.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * OWASP Enterprise Security API (ESAPI)
+ *
+ * This file is part of the Open Web Application Security Project (OWASP)
+ * Enterprise Security API (ESAPI) project. For details, please see
+ * <a href="http://www.owasp.org/index.php/ESAPI">http://www.owasp.org/index.php/ESAPI</a>.
+ *
+ * Copyright (c) 2007 - The OWASP Foundation
+ *
+ * The ESAPI is published by OWASP under the BSD license. You should read and accept the
+ * LICENSE before you use, modify, and/or redistribute this software.
+ *
+ * @author Jeff Williams <a href="http://www.aspectsecurity.com">Aspect Security</a>
+ * @created 2007
+ */
+package org.owasp.esapi.logging.log4j;
+
+import org.apache.log4j.Priority;
+import org.apache.log4j.spi.LoggerFactory;
+import org.owasp.esapi.ESAPI;
+import org.owasp.esapi.logging.appender.LogAppender;
+import org.owasp.esapi.logging.cleaning.LogScrubber;
+import org.owasp.esapi.reference.DefaultSecurityConfiguration;
+
+/**
+ * Implementation of the LoggerFactory interface. This implementation has been 
+ * overridden to return instances of org.owasp.esapi.reference.Log4JLogger.
+ *
+ * @author August Detlefsen (augustd at codemagi dot com)
+ *         <a href="http://www.codemagi.com">CodeMagi, Inc.</a>
+ * @since October 15, 2010
+ * @see org.owasp.esapi.logging.log4j.Log4JLogFactory
+ * @see org.owasp.esapi.reference.Log4JLogger
+ */
+@Deprecated
+public class Log4JLoggerFactory implements LoggerFactory {
```

Comment:
This class is missing tests.

Reply:
@kwwall  @xeno6696, I've added a very basic test class to this implementation.  It is not very well covered, but it does assert that the appender/scrubber will be applied to log content in a happy-path scenario for the default Log4J levels.

Please let me know if extending these tests would be required in this effort.

https://github.com/ESAPI/esapi-java-legacy/pull/533/commits/1e930c446e8276e9050e0ffc2ed56fe062c899f7

Reply:
@xeno6696  @kwwall I believe this is the only additional outstanding issue that requires feedback.